Support paper crawl endpoints

andyhaskell opened the pull request:
This PR adds support for the Dropbox Paper `docs/list(/continue)?`, download, folder info, and alpha get metadata endpoints

andyhaskell committed:
Add support for Dropbox Paper list, download, get folder info methods

andyhaskell committed:
Add support for Dropbox Paper's alpha get metadata endpoint

andyhaskell committed:
Add test coverage for Dropbox Paper API

Additionally, add subdomain parameter to base download method; files
endpoints download from content.dropboxapi.com, while Dropbox Paper's
download endpoint is on api.dropboxapi.com. This commit also adds a
Paper API to the Client struct.

kujenga commented on paper_test.go:
```diff
@@ -0,0 +1,113 @@
+	Revision        int64          `json:"revision"`
+	LastUpdatedDate time.Time      `json:"last_updated_date"`
+	LastEditor      string         `json:"last_editor"`
+}*/
```
What’s the purpose of this struct?

andyhaskell replied:
That was commented out; thought my autocomplete was down but then I just realized I hadn't added `.Paper` wasn't on the `Client` struct

kujenga commented on paper_test.go:
```diff
@@ -0,0 +1,113 @@
+	c := client()
+
+	var fileID string
+	if fileID = os.Getenv("DROPBOX_PAPER_TOP_LEVEL_FILE_ID"); fileID == "" {
```
How does one get this ID? Can we make it automatic?

andyhaskell replied:
Those you just pick yourself, just like the token envionment variable used in in `client()`

andyhaskell replied:
I was tempted to implement the `Upload` endpoint and throw that into the test, but for that you'd still need the identifier for a folder if you want to upload a Paper in a folder, then retrieve it, since there's no create folder method in the Dropbox HTTP docs currently

kujenga reviewed (approved):
LGTM! A few thoughts on the tests.

andyhaskell committed:
Add Dropbox Paper convenience method for making throwaway Paper files

This is run at the beginning of tests for the get_metadata,
get_folder_info, and download endpoints so that a user running the
test doesn't have to set an environment variable to the ID of one of
their Papers, outside of the TestPaper_GetFolderInfoInsideFolders test,
since there isn't an easy way to make a throwaway file that is inside
a folder.

For these tests that create throwaway files, the environment variable
TEST_CAN_CREATE_DBOX_PAPER must be set so a user knows that they are
running tests that do writes to their Dropbox.
